### PR TITLE
[Functionalization] Fix more cpp tests that require new lowerings

### PR DESCRIPTION
**Pull request by wonjoolee95**
Fix rest of the cpp tests that requires new lowerings. These include:
- SLogDet
- Inverse
- Pinverse

### CODE REVIEW

**Comment by wonjoolee95** [comment]
The failures seem real, the `test_ops.py` for the updated ops are failing. Working on a fix.

**Comment by wonjoolee95** [comment]
Okay, the `test_ops.py` still seem to give some trouble. For example the `AllowedOpInfoEntry('linalg.svd')` fails with error:

> ```2023-01-25 05:56:04.965700: F 2566733 /workspace/pytorch/xla/third_party/tensorflow/bazel-tensorflow/tensorflow/compiler/xla/xla_client/debug_macros.h:27] Non-OK-status: status.status() status: INVALID_ARGUMENT: Slice dim size 1 greater than dynamic slice dimension: 0.```

It looks the exact root cause of this failure is at https://github.com/pytorch/xla/blob/master/torch_xla/csrc/ops/svd.cpp#L19. In the `ShapeOfXlaOp` call, this error is raised. The input that causes this error seems to be:
```
xla_tensor=1
full_matrices=0
compute_uv=1
```

**Comment by wonjoolee95** [comment]
The newly failing `test/test_ops.py` ops include:
- `AllowedOpInfoEntry('geqrf')`
- `AllowedOpInfoEntry('linalg.det')`
- `AllowedOpInfoEntry('linalg.eig')`
- `AllowedOpInfoEntry('linalg.inv')`
- `AllowedOpInfoEntry('linalg.matrix_rank')`
- `AllowedOpInfoEntry('linalg.svd')`
- `AllowedOpInfoEntry('linalg.svdvals')`

I've commented out just to let the CI run for now.

Many of them (including `geqrf` op which seemingly doesn't appear related) fail at `SVD::Lower` like:
```
	tsl::CurrentStackTrace[abi:cxx11]()
	xla::Shape const* ConsumeValue<xla::Shape const*>(tsl::StatusOr<xla::Shape const*>&&)
	torch_xla::XlaHelpers::ShapeOfXlaOp(xla::XlaOp)
	
	torch_xla::SVD::Lower(torch_xla::LoweringContext*) const
```
This is the same error described in https://github.com/pytorch/xla/pull/4496#issuecomment-1403202882.

**Comment by wonjoolee95** [comment]
The `_linalg_svd` *should* be fixed by https://github.com/pytorch/xla/pull/4537. I'll rebase once that PR merges.

**Comment by wonjoolee95** [comment]
https://github.com/pytorch/xla/pull/4537 is merged with master, so some of the errors that we were seeing *should* be gone. While I test this locally, re-running CI to also verify.

**Comment by wonjoolee95** [comment]
Okay, locally at least `test/test_ops.py` is all passing. I'll let CI verify the rest.

**Comment by wonjoolee95** [comment]
CPU CI passing, will still wait for the GPU CI but this should be finally ready for review.

File: torch_xla/csrc/aten_xla_type.cpp
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:1560]: It looks like we didn't do anything with `check_errors`. An easy fix for now is to fallback to cpu if check_errors is provided. wdyt?
https://pytorch.org/docs/stable/generated/torch.linalg.inv_ex.html

**Review by alanwaketan — COMMENTED** [review]
Generally LGTM. Left one comment.

File: torch_xla/csrc/aten_xla_type.cpp
Comment (wonjoolee95) [reply torch_xla/csrc/aten_xla_type.cpp:1560]: Good point, makes sense. According to https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L13160, the default value for `check_errors` is `false`, so I've updated to fallback to cpu when this is set to `true`. Also added a comment explaining this in the code.

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Pull request merged**
